### PR TITLE
Handle case when b2c profile contains a '.'

### PR DESCRIPTION
**Pull request by neha-bhargava**
#2444 

Try to handle the scenario when the b2c profile have a '.' character.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AccountId.cs [hunk 1]
```diff
@@ -63,14 +63,14 @@ internal static AccountId ParseFromString(string str)
...
+            int lastIndexOfDot = str.LastIndexOf('.');
 
-            if (elements.Length == 1)
+            if (lastIndexOfDot == -1)
             {
                 return new AccountId(str); //Account id is from ADFS; no . in the string
             }
 
-            return new AccountId(str, elements[0], elements[1]);
+            return new AccountId(str, str.Substring(0, lastIndexOfDot), str.Substring(lastIndexOfDot + 1));
```
Comment (neha-bhargava) [inline src/client/Microsoft.Identity.Client/AccountId.cs:73]: This is the fix, splitting the string based on the last `.`

File: src/client/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 2]
```diff
@@ -117,7 +117,7 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync(string userFlow)
             var accounts = await GetAccountsInternalAsync(ApiIds.GetAccountsByUserFlow).ConfigureAwait(false);
 
             return accounts.Where(acc =>
-                acc.HomeAccountId.ObjectId.Split('.')[0].EndsWith(
+                acc.HomeAccountId.ObjectId.EndsWith(
```
Comment (neha-bhargava) [inline src/client/Microsoft.Identity.Client/ClientApplicationBase.cs:120]: I don't see the reason for adding this split here as the ObjectId is obtained after splitting the homeAccountId with the `.`

File: src/client/Microsoft.Identity.Client/AccountId.cs (hunk 1 above)
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AccountId.cs:73]: Very cool!

Can you add more tests around these string operations? Lots of "off by 1" errors possible. This method is static so should be very easy to test.

**Review by bgavrilMS — APPROVED** [review]
LGTM, please add a few unit tests to cover the parsing

File: src/client/Microsoft.Identity.Client/AccountId.cs (hunk 1 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/AccountId.cs:73]: Added tests for parsing

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/AccountIdTest.cs [hunk 3]
```diff
@@ -34,5 +34,75 @@ public void AccountIdPublicApi()
...
+
+        [TestMethod]
+        [Description("Test the parsing of HomeAccountId with no dot")]
+        public void ParseFromStringNoDotTest()
+        {
+            AccountId accountId = AccountId.ParseFromString("uid");
+
+            Assert.AreEqual("uid", accountId.ObjectId);
+            Assert.AreEqual(null, accountId.TenantId);
+        }
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/AccountIdTest.cs:46]: The tests can be condensed into one test with adding `[DataRow]` attribute like in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cfa2b36b9b5732d182bf6e31bb8a9fbc1127ab3a/tests/Microsoft.Identity.Test.Unit/AppConfigTests/PublicClientApplicationBuilderTests.cs#L383

**Review by pmaytak — APPROVED** [review]
👍

**Pull request merged**